src/pages/create-or-edit-lesson/CreateOrEditLesson.tsx

```diff
@@ -165,44 +155,71 @@ const CreateOrEditLesson = () => {
   } = useForm<LessonData>({
     initialValues,
     validations,
-    onSubmit: id ? fetchEditedLesson : fetchAddLesson,
+    onSubmit: () => {
+      if (id) {
+        fetchEditedLesson()
+      } else {
+        fetchAddLesson()
+      }
+    },
     submitWithData: true
   })
 
-  const getLesson = (id?: string): Promise<AxiosResponse> => {
-    return ResourceService.getLesson(id)
-  }
+  const getLesson = useCallback(() => {
+    if (id) {
+      return ResourceService.getLesson(id)
+    }
 
-  const handleResponseLesson = (lesson: LessonData) => {
-    for (const key in data) {
-      const validKey = key as keyof LessonData
-      handleNonInputValueChange(validKey, lesson[validKey])
+    return defaultResponse
+  }, [id])
+
+  const {
+    isLoading,
+    error,
+    data: lesson
+  } = useQuery({
+    queryKey: ['lesson', id],
+    queryFn: getLesson,
+    options: {
+      initialData: defaultResponse,
+      enabled: Boolean(id)
     }
-  }
+  })
+
+  const editLesson = useCallback(async () => {
+    if (id) {
+      return ResourceService.editLesson(data, id)
+    }
+
+    return Promise.resolve(defaultResponse)
+  }, [data, id])
 
-  const { loading: getLessonLoading, fetchData: fetchDataLesson } = useAxios<
-    LessonData,
-    string
-  >({
-    service: getLesson,
-    fetchOnMount: false,
-    defaultResponse,
-    onResponse: handleResponseLesson,
-    onResponseError: handleResponseError
+  const { mutate: fetchEditedLesson } = useMutation({
+    mutationFn: editLesson,
+    onSuccess: handleResponse,
+    onError: handleResponseError
   })
 
   useEffect(() => {
-    if (id) {
-      void fetchDataLesson(id)
+    if (lesson && id) {
+      for (const key in data) {
+        const validKey = key as keyof LessonData
+        handleNonInputValueChange(validKey, lesson[validKey])
+      }
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [id])
+  }, [lesson, id])
+
+  useEffect(() => {
+    if (error) {
+      handleResponseError(error)
+    }
+  }, [error, handleResponseError])
 
-  if (getLessonLoading) {
+  if (isLoading) {
     return <Loader pageLoad />
   }
 
-  const attachmentsList = data.attachments.map((attachment) => (
+  const attachmentsList = data?.attachments?.map((attachment) => (
```

Comment:
Also, please, remove `?` here
```suggestion
  const attachmentsList = data.attachments?.map((attachment) => (
```